### PR TITLE
feat: creating extension configuration registry

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Creating a dedicated registry to hold configuration (preferences) logic.

### Notable changes

- ConfigurationRegistry manage the `ai-lab.models.path` previously managed by `ModelsManager`
- Creating frontend store holding the configuration values

### How to test this PR?

- [x] Unit tests has been provided

### CODE REVIEW

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Should be splut in 2 PRs as it mixes 2 different concerns

**Comment by axel7083** [comment]
> Should be splut in 2 PRs as it mixes 2 different concerns

@jeffmaury I updated the PR to only focus on the creation of the ConfigurationRegistry

File: packages/backend/src/registries/ConfigurationRegistry.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { configuration, type Configuration, type Disposable, type Webview } from '@podman-desktop/api';
+import { Publisher } from '../utils/Publisher';
+import type { ExtensionConfiguration } from '@shared/src/models/IExtensionConfiguration';
+import { Messages } from '@shared/Messages';
+import path from 'node:path';
+
+const CONFIGURATION_SECTIONS: string[] = ['ai-lab.experimentalGPU', 'ai-lab.models.path'];
```
Comment (jeffmaury) [inline packages/backend/src/registries/ConfigurationRegistry.ts:24]:
```suggestion
const CONFIGURATION_SECTIONS: string[] = ['ai-lab.models.path'];
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**